mlflow/types/schema.py

```diff
@@ -122,34 +127,401 @@ def from_numpy_type(cls, np_type):
+        try:
+            self._dtype = DataType[dtype] if isinstance(dtype, str) else dtype
+        except KeyError:
+            raise MlflowException(
+                f"Unsupported type '{dtype}', expected instance of DataType, Array, Object or "
+                f"one of {[t.name for t in DataType]}"
+            )
+        if not isinstance(self.dtype, (DataType, Array, Object)):
+            raise MlflowException(
+                "Expected mlflow.types.schema.Datatype, mlflow.types.schema.Array, "
+                "mlflow.types.schema.Object or str for the 'dtype' "
+                f"argument, but got {self.dtype.__class__}"
+            )
+        self._required = required
+
+    @property
+    def name(self) -> str:
+        """The property name."""
+        return self._name
+
+    @property
+    def dtype(self) -> Union[DataType, "Array", "Object"]:
+        """The property data type."""
+        return self._dtype
+
+    @property
+    def required(self) -> bool:
+        """Whether this property is required"""
+        return self._required
+
+    @required.setter
+    def required(self, value: bool) -> None:
+        self._required = value
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, Property):
+            return (
+                self.name == other.name
+                and self.dtype == other.dtype
+                and self.required == other.required
+            )
+        return False
+
+    def __lt__(self, other) -> bool:
+        return self.name < other.name
+
+    def to_dict(self):
+        d = {"type": self.dtype.name} if isinstance(self.dtype, DataType) else self.dtype.to_dict()
+        d["required"] = self.required
+        return {self.name: d}
+
+    @classmethod
+    def from_json_dict(cls, **kwargs):
+        """
+        Deserialize from a json loaded dictionary.
+        The dictionary is expected to contain only one key as `name`, and
+        the value should be a dictionary containing `type` and
+        optional `required` keys.
+        Example: {"property_name": {"type": "string", "required": True}}
+        """
+        if len(kwargs) != 1:
+            raise MlflowException(
+                f"Expected Property JSON to contain a single key as name, got {len(kwargs)} keys."
+            )
+        name, dic = kwargs.popitem()
+        if not {"type"} <= set(dic.keys()):
+            raise MlflowException(f"Missing keys in Property `{name}`. Expected to find key `type`")
+        required = dic.pop("required", True)
+        dtype = dic["type"]
+        if dtype not in [ARRAY_TYPE, OBJECT_TYPE]:
+            return cls(name=name, dtype=dtype, required=required)
+        if dtype == ARRAY_TYPE:
+            return cls(name=name, dtype=Array.from_json_dict(**dic), required=required)
+        if dtype == OBJECT_TYPE:
+            return cls(name=name, dtype=Object.from_json_dict(**dic), required=required)
+
+    def _merge_property(self, prop: "Property") -> "Property":
```

Comment:
```suggestion
    def _merge(self, prop: "Property") -> "Property":
```

Do we need `_property`?